library/network/src/modules/NetworkInterfaces.rb

```diff
@@ -1299,73 +1300,66 @@ def GetDevTypeDescription(type, longdescr)
     # @return dumped settings (later acceptable by Import())
     def Export(devregex)
       devs = Filter(@Devices, devregex)
-      Builtins.y2debug("Devs=%1", devs)
-      Convert.convert(devs, from: "map", to: "map <string, map>")
+      log.debug("Devs=#{devs}")
+      devs
     end
 
     # Were the devices changed?
     # @return true if modified
     def Modified(devregex)
       devs = Filter(@Devices, devregex)
       original_devs = Filter(@OriginalDevices, devregex)
-      Builtins.y2debug("OriginalDevs=%1", original_devs)
-      Builtins.y2debug("Devs=%1", devs)
+      log.debug("OriginalDevs=#{original_devs}")
+      log.debug("Devs=#{devs}")
       devs == original_devs
     end
 
     def GetFreeDevices(type, num)
-      Builtins.y2debug("Devices=%1", @Devices)
-      Builtins.y2debug("type,num=%1,%2", type, num)
-      Builtins.y2debug("Devices[%1]=%2", type, Ops.get(@Devices, type, {}))
+      log.debug("Devices=#{@Devices}")
+      log.debug("type,num=#{type},#{num}")
+      log.debug("Devices[#{type}]=#{@Devices[type]}")
 
-      curdevs = []
-      Builtins.foreach(
-        Convert.convert(
-          Map.Keys(Ops.get(@Devices, type, {})),
-          from: "list",
-          to:   "list <string>"
-        )
-      ) do |dev|
-        dev = device_num(dev) if Builtins.issubstring(dev, type)
-        curdevs = Builtins.add(curdevs, dev)
-      end
+      curdevs = @Devices.fetch(type, {}).keys
+      curdevs = curdevs.map { |d| d.include?(type) ? device_num(d) : d }
 
       i = 0
       count = 0
       ret = []
 
       # Hotpluggable devices
-      if IsHotplug(type) && !Builtins.contains(curdevs, "")
-        Builtins.y2debug("Added simple hotplug device")
-        count = Ops.add(count, 1)
-        ret = Builtins.add(ret, "")
+      if IsHotplug(type) && !curdevs.include?("")
+        log.debug("Added simple hotplug device")
+        count += 1
+        ret << ""
       end
 
       # Remaining numbered devices
-      while Ops.less_than(count, num)
-        ii = Builtins.sformat("%1", i)
-        if !Builtins.contains(curdevs, ii)
-          ret = Builtins.add(ret, ii)
-          count = Ops.add(count, 1)
+      while count < num
```

Comment:
I doesn't look like proper ruby. What about using `(count...num).each`? You could also get rid of the `i` counter.